.github/workflows/pre-commit.yml

```
@@ -7,9 +7,9 @@ jobs:
   pre-commit:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9
+      - uses: actions/checkout@checkout@v3
         with:
           fetch-depth: 0
-      - uses: pre-commit/action@646c83fcd040023954eafda54b4db0192ce70507
+      - uses: pre-commit/action@action@v3.0.1
```

typo 
```suggestion
      - uses: pre-commit/action@v3.0.1
```

While these are probably safe, we've used the hash version and have let dependabot update them as new versions come out.  The risk is that break changes or malicious versions could be merged to those third party repos, and we'd use those versions automatically

---

Good point